### DISCUSSION TITLE
Is there any logging that can be enabled to catch issues opening MusicXML files?

### DISCUSSION THREAD

**Discussion by tledford4** [musescore/MuseScore#14374] [General]
I noticed in the MuseScore forums that github was the place to ask questions that probably are best answered by developers as they were more likely to see them here.

I've been exporting MusicXML from another notation program (Notion) and opening them in MuseScore 4, then saving them in native MuseScore format to see how that goes.  So far, it has been very encouraging, but there have been a handful of larger files (>5MB) that, after 30 seconds or so, return an error "File '' is corrupted" with a Cancel and Ignore button.  Pressing Ignore results in a Force Quit/Wait loop ("MuseScore is not responding") until either MuseScore is killed or Cancel is selected.

This behavior occurs on Ubuntu 22.04 and 22.10 and Manjaro (rolling release, latest updates applied).  Haven't tried it on Windows yet but will.

Is there any logging in place that is done during a file open, or can some sort of debug logging be enabled by any chance?  It might be that whatever is making MuseScore choke could easily be fixed in the Notion source file and then it be re-exported.

Hope this makes sense, and please let me know if you need any other information.

**Comment by MarcSabatella** [comment]
If you're on Linux, presumably you are running the AppImage.  So just start it from a terminal, and you'll see any error messages logged there.  No idea if that will provide useful here, but definitely I've seen some info there that helped me track down few problems in various MuseScore 4 builds.

**Reply by tledford4** [reply]
Thank you, Marc, I should have thought of that, and it DID yield potentially useful information, complaining about something unexpected in measure 126 staff number 28.  I also discovered the debug flag '-d' and testing again with debug enabled, but the same error info was received at the "choke point."

At least now I can examine the source document more closely to see if it can be cleaned up, or determine if Notion's MusicXML export is borked.

Thanks again!
